connect-inject/handler.go

```diff
@@ -169,12 +170,21 @@ type Handler struct {
 	// Only necessary if ACLs are enabled.
 	CrossNamespaceACLPolicy string
 
-	// Default resource settings for sidecar proxies.
+	// Default resource settings for sidecar proxies. Some of these
+	// fields may be empty.
 	DefaultProxyCPURequest    resource.Quantity
 	DefaultProxyCPULimit      resource.Quantity
 	DefaultProxyMemoryRequest resource.Quantity
 	DefaultProxyMemoryLimit   resource.Quantity
```

Comment:
I think it'd be nice to make these into `corev1.ResourceRequirements` for consistency.

Reply:
I looked at this, but it gets complicated because there are annotations associated with these settings and they're allowed to be empty. This is significantly different behavior to the init and lifecycle resource flags so it is reasonable to treat them differently. Also, given that it is not a trivial change, it is scope creep because it's not critical to the addition of init and lifecycle resource flags.

Reply:
I agree!
It gets a little messy dealing with ones that are not specified though :-/